kotlinx-coroutines-core/common/src/flow/terminal/Collect.kt

```diff
@@ -86,9 +86,45 @@ public suspend inline fun <T> Flow<T>.collectIndexed(crossinline action: suspend
         override suspend fun emit(value: T) = action(checkIndexOverflow(index++), value)
     })
 
+/**
+ * Terminal flow operator that collects the given flow with a provided [action].
+ * The crucial difference from [collect] is that when the original flow emits a new value, [action] block for previous
+ * value is cancelled.
+ * It can be demonstrated by the following example:
```

Comment:
Makes sense to end the summary here.

Reply:
What do you mean by "summary"?

Reply:
kdoc has summary and details parts separated by an empty line. The example should go in details.